samba/CHANGELOG.md

```diff
@@ -1,4 +1,14 @@
 # Changelog
+## 13.0.0
+
+BREAKING CHANGE: Default shares will be limited to "media" and "share" by default.  
+
+This change enhances security and user control, but may require updating your configuration
+to access shares that contain potentially sensitive information.
+
+- Default shares reduced to 'media' and 'share'
```

Comment:
The logical reasoning behind this is unclear to me. The biggest use case for this add-on is accessing backups and the Home Assistant configuration.

Reply:
I'd like to add that the current defaults already limit shares to local networks, and require authentication. So reducing the default shares would limit use cases for minimal security benefits (only if home network and samba authentication are both compromised).

Reply:
I fail to see that logic in this case, as it assumes those shares contain no sensitive information, which is very unlikely.

Additionally, the main use case for this add-on is accessing one's Home Assistant configuration; the add-on should be set up for that use case out of the box.

Reply:
I agree with everything you are saying on both the defaults, as well as the use cases.  However, the counterpoints I would make are:

- exposing a share by default that allows either 'nuking' HA (by removing or encrypting all containing files - like config.yaml) or capturing secrets feels like it should be something the user should choose
- While we are protected by default to local network, and username/password authentication, that assumes that the attack on it will be direct from the internet.
- Tautological - for a user to use Samba, they have to connect to it using SMB, and authenticate to it.
- In a world of viruses and ransomware, having the share enabled (used or not), and having that connection from the computer is enough for anything that attacks the users PC to 'make the jump' to also impacting HA.  If the users computer is infected, and they have connected to SMB on HA, the only things safe are the things not exposed.

When I can access files such as
- '.jwt_secret'
- secrets.yaml
- the private key for the certificate for HA Cloud
- the JWT for the ACME account for it
All falling under what is exposed using default filters in just the config folder, and know at least most of them are in the backups as well - then allowing access to them as the default case feels wider than it should be .

I feel strongly about this, but do not view most of it as a 'hill to die on'.  If you disagree that the protection provided by only exposing things that are innocuous by default is worth the additional hassle of users having to turn on the specific shares they want, I will update, as the option is a step better than today, even if the defaults expose more than is ideal.  

With that said, I would view the SSL folder as being a very high bar for saying 'we should be exposing this by default'.  And, if we are exposing more of the other shares by default, I will want to update to add a few more default veto_files (covering at least the highest profile of the ones listed above).

Reply:
Hi, I'm a user interested on seeing this happen.

From this thread, I understand the only drawback was the selection of the 'default enabled' options? In which case I guess the PR dev can easily set the /config and /backup shares as default to support the "accessing one's Home Assistant configuration; or transferring backups" default use case.

Would that unlock this PR to be merged?

Reply:
This PR has been discussed internally today; the conclusion is that we won't be accepting it in its current state.

We are happy to add the options to enable/disable access to certain shares. However, we will not be accepting new defaults on which of those shares are enabled or disabled at this point.

Some reasoning:

- Disabling certain shares, the way it is done in this PR, will be a breaking change for existing users (as they suddenly will lose access to shares).
- The main goal of why this add-on was created is to be able to access your configuration. This PR defeats that purpose, meaning every user starting with this add-on will run into oddities, not understanding why they can't see that. This creates additional frustration on the initial setup (especially for newer users).

Ideally, we agree, we would love to offer and force a choice to the user (similar like we do with the password setting), however, we don't have the UX abilities to make that happen for this specific case at the moment.

For the above reasoning, we welcome the new feature to disable shares, but we are not willing to accept a new default.

../Frenck

Reply:
TL;DR this is ready for review

In a number of ways, my refactor turned into a rewrite, but I believe that it has accomplished the requested changes, while also accomplishing the minimum goals:
- No breaking change is introduced
- No change of functionality without end user reconfiguration for a current install
- Users can opt to turn off specific shares
- No config introduced requires YAML config by the end user

In addition, the update had two further goals:
- config is organized in such a way that, several months from now, the config ***could*** be updated in another PR to change the default without affecting existing installs, and make that a viable 'change before first run' (like the password, as mentioned by @frenck would be the 'ideal case')
- config required is condensed as much as possible and is as high in the configuration screen as reasonable (so if that happens, the config is on the first screen and near the password field)

Looking for any feedback for the path to getting this PR merged from here.